cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -94,99 +103,195 @@ public void evaluate() throws Throwable {
     );
-    NamespaceMeta updated = new NamespaceMeta.Builder(meta).setDescription("new desc").build();
-    namespaceAdmin.updateProperties(namespace.toId(), updated);
-    namespaceAdmin.delete(namespace.toId());
-    // once the namespace has been deleted, privileges on that namespace should be revoked
+    appManager.delete();
+    // All privileges on the app should have been revoked
+    Assert.assertEquals(
+      ImmutableSet.of(
+        new Privilege(instance, Action.ADMIN),
+        new Privilege(AUTH_NAMESPACE, Action.ALL)
+      ),
+      authorizer.listPrivileges(ALICE)
+    );
+    // Deploy a couple of apps in the namespace
+    deployApplication(AUTH_NAMESPACE.toId(), AllProgramsApp.class);
+    deployApplication(AUTH_NAMESPACE.toId(), WorkflowApp.class);
+    Assert.assertEquals(
+      ImmutableSet.of(
+        new Privilege(instance, Action.ADMIN),
+        new Privilege(AUTH_NAMESPACE, Action.ALL),
+        new Privilege(AUTH_NAMESPACE.app(AllProgramsApp.NAME), Action.ALL),
+        new Privilege(AUTH_NAMESPACE.app(WorkflowApp.class.getSimpleName()), Action.ALL)
+      ),
+      authorizer.listPrivileges(ALICE)
+    );
+    // revoke all privileges on an app.
+    authorizer.revoke(AUTH_NAMESPACE.app(WorkflowApp.class.getSimpleName()));
     Assert.assertEquals(
       ImmutableSet.of(
-        new Privilege(NamespaceId.DEFAULT, Action.ADMIN),
-        new Privilege(instance, Action.ADMIN)
+        new Privilege(instance, Action.ADMIN),
+        new Privilege(AUTH_NAMESPACE, Action.ALL),
+        new Privilege(AUTH_NAMESPACE.app(AllProgramsApp.NAME), Action.ALL)
+      ),
+      authorizer.listPrivileges(ALICE)
+    );
+    // deleting all apps should fail because alice does not have admin privileges on the Workflow app
+    try {
+      deleteAllApplications(AUTH_NAMESPACE);
+      Assert.fail("Deleting all applications in the namespace should have failed because alice does not have ADMIN " +
+                    "privilege on the workflow app.");
+    } catch (UnauthorizedException expected) {
+      // expected
+    }
+    // grant admin privilege on the WorkflowApp. deleting all applications should succeed.
+    authorizer.grant(AUTH_NAMESPACE.app(WorkflowApp.class.getSimpleName()), ALICE, ImmutableSet.of(Action.ADMIN));
+    Assert.assertEquals(
+      ImmutableSet.of(
+        new Privilege(instance, Action.ADMIN),
+        new Privilege(AUTH_NAMESPACE, Action.ALL),
+        new Privilege(AUTH_NAMESPACE.app(WorkflowApp.class.getSimpleName()), Action.ADMIN)
+      ),
+      authorizer.listPrivileges(ALICE)
+    );
+    deleteAllApplications(AUTH_NAMESPACE);
+    // deleting all apps should remove all privileges on all apps, but the privilege on the namespace should still exist
+    Assert.assertEquals(
+      ImmutableSet.of(
+        new Privilege(instance, Action.ADMIN),
+        new Privilege(AUTH_NAMESPACE, Action.ALL)
       ),
-      authorizer.listPrivileges(alice)
+      authorizer.listPrivileges(ALICE)
```

Comment:
Nice tests! Do we cover the case where User BOB deploys an App and we assert that it doesn't show up in ALICE's privileges? Or will it since she has ALL access on the AUTH_NAMESPACE?